core/src/main/java/org/elasticsearch/index/query/QueryBuilders.java

```diff
@@ -277,8 +277,8 @@ public static SimpleQueryStringBuilder simpleQueryStringQuery(String queryString
      * Unlike the "NOT" clause, this still selects documents that contain undesirable terms,
      * but reduces their overall score:
      */
-    public static BoostingQueryBuilder boostingQuery() {
-        return new BoostingQueryBuilder();
+    public static BoostingQueryBuilder boostingQuery(QueryBuilder positiveQuery, QueryBuilder negativeQuery) {
```

Comment:
I was carried away by making inner queries final and removing the no-args constructor. After refactoring #11629 I realize that at least for now we need support for setting inner queries to `null`, so I will revert this.